src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Performance/PreferHashDataOverComputeHashTests.cs

```diff
@@ -0,0 +1,1188 @@
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        byte[] digest;
+        using ({|#2:SHA256 sha256 = SHA256.Create()|})
+        {
+            int aboveLine = 20;
+            digest = {|#0:sha256.ComputeHash({|#1:buffer|})|};
+            int belowLine = 10;
+        }
+    }
+}
+";
+
+            string csFix = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        byte[] digest;
+        int aboveLine = 20;
+        digest = SHA256.HashData(buffer);
+        int belowLine = 10;
+    }
+}
+";
+            await TestCSWithVariable(
+                csInput,
+                csFix,
+                "System.Security.Cryptography.SHA256");
+        }
+
+        [Fact]
+        public async Task BasicCreateHelperUsingBlockCase2()
+        {
+            string vbInput = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim digest As Byte()
+        {|#2:Using sha256 As SHA256 = SHA256.Create()|}
+            Dim aboveLine = 20
+            digest = {|#0:sha256.ComputeHash({|#1:buffer|})|}
+            Dim belowLine = 10
+        End Using
+    End Sub
+End Class
+";
+
+            string vbFix = @"
+Imports System
+Imports System.Security.Cryptography
+
+Public Class Test
+    Public Shared Sub TestMethod()
+        Dim buffer = New Byte(1023) {}
+        Dim digest As Byte()
+        Dim aboveLine = 20
+        digest = SHA256.HashData(buffer)
+        Dim belowLine = 10
+    End Sub
+End Class
+";
+            await TestVBWithVariable(
+                vbInput,
+                vbFix,
+                "System.Security.Cryptography.SHA256");
+        }
+
+        [Fact]
+        public async Task CSharpCreateHelperUsingStatementCastedCase()
+        {
+            string csInput = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        using ({|#2:HashAlgorithm sha256 = SHA256.Create()|})
+        {
+            int aboveLine = 20;
+            byte[] digest = {|#0:sha256.ComputeHash({|#1:buffer|})|};
+            int belowLine = 10;
+        }
+    }
+}
+";
+
+            string csFix = @"
+using System;
+using System.Security.Cryptography;
+
+public class Test
+{
+    public static void TestMethod()
+    {
+        var buffer = new byte[1024];
+        int aboveLine = 20;
+        byte[] digest = SHA256.HashData(buffer);
+        int belowLine = 10;
+    }
+}
+";
+            await TestCSWithVariable(
+                csInput,
+                csFix,
+                "System.Security.Cryptography.SHA256");
+        }
+
+        [Fact]
+        public async Task BasicCreateHelperUsingBlockCastedCaseNoFix()
+        {
+            // unable to get type info for SHA256
```

Comment:
Something has changed since this test was written. The typeinfo can now be obtained.